packages/front-end/components/Settings/BigQueryForm.tsx

```diff
@@ -1,15 +1,66 @@
-import { ChangeEventHandler, FC } from "react";
+import { ChangeEventHandler, FC, useState } from "react";
 import { BigQueryConnectionParams } from "back-end/types/integrations/bigquery";
 import { isCloud } from "@/services/env";
+import { useAuth } from "@/services/auth";
 import Field from "../Forms/Field";
 import Tooltip from "../Tooltip/Tooltip";
+import SelectField from "../Forms/SelectField";
+import Button from "../Button";
 
 const BigQueryForm: FC<{
   params: Partial<BigQueryConnectionParams>;
   existing: boolean;
   setParams: (params: { [key: string]: string }) => void;
   onParamChange: ChangeEventHandler<HTMLInputElement | HTMLSelectElement>;
 }> = ({ params, setParams, existing, onParamChange }) => {
+  const [testConnectionResults, setTestConnectionResults] = useState<{
+    status: "success" | "danger" | "warning";
+    message: string;
+    datasetOptions: string[];
+  } | null>(null);
+  const [loadingDatasetOptions, setLoadingDatasetOptions] = useState(false);
+  const { apiCall } = useAuth();
+
+  async function testConnection() {
+    try {
+      setTestConnectionResults(null);
+      const { datasets } = await apiCall<{ datasets: string[]; error: string }>(
+        "/datasources/fetch-bigquery-datasets",
+        {
+          method: "POST",
+          body: JSON.stringify({
+            projectId: params.projectId,
+            client_email: params.clientEmail,
+            private_key: params.privateKey,
+          }),
+        }
+      );
+
+      if (!datasets.length) {
+        setTestConnectionResults({
+          status: "warning",
+          datasetOptions: [],
+          message:
+            "We were able to connect to BigQuery, but we weren't able to retreive any datasets in this project.",
+        });
+        return;
+      }
+      setTestConnectionResults({
+        status: "success",
+        datasetOptions: datasets,
+        message: `Connected to ${params.projectId} successfully!`,
+      });
+      //TODO: Intelligently select a default dataset based on the datasource type (manual, ga4, segment, etc)
```

Comment:
It's worth doing something really simple here now.  Better than nothing and we can improve it over time.

```ts
const analyticsDataset = datasets.find(d => d.match(/^analytics_/));
if (analyticsDataset) {
  setParams({ ["defaultDataset"]: analyticsDataset })
}
```